packages/ckeditor5-engine/src/model/utils/insertcontent.js

```diff
@@ -440,9 +437,20 @@ class Insertion {
 	/**
 	 * Merges sibling of the first node if should be merged.
 	 *
+	 * After the content was inserted we may try to merge it with its siblings.
+	 * This should happen only if the selection was in those elements initially.
+	 *
+	 * Example:
+	 * 		<p>x^</p> + <p>y</p> => <p>x</p><p>y</p> => <p>xy[]</p>
+	 * 		and:
+	 * 		<p>x^y</p> + <p>z</p> => <p>x</p>^<p>y</p> + <p>z</p> => <p>x</p><p>z</p><p>y</p> => <p>xz[]y</p>
+	 * 		but:
+	 * 		<p>x</p><p>^</p><p>z</p> + <p>y</p> => <p>x</p><p>y</p><p>z</p> (no merging)
+	 * 		<p>x</p>[<img>]<p>z</p> + <p>y</p> => <p>x</p><p>y</p><p>z</p> (no merging, note: after running deleteContents
```

Comment:
I still don't think this should be here. This is just incorrect example, it shouldn't be here. `Insertion` class does not operate on a range.

I think this whole comment (I mean all three examples without the last) should be somewhere else (`handleNodes()`? or `Insertion` class description?).